designdocs/user-handles.md

```diff
@@ -0,0 +1,343 @@
+# Frequency  Handles
+
+## Context and Scope
+
+The Frequency blockchain aims to provide a mechanism to register and retrieve user handles on chain to make it easier to use MSA Ids.
+To achieve this, we propose the creation of a user handle ```registry``` system on Frequency chain, coupled with ```MessageSourceAccount```, that allows users to choose a handle for their MSA Ids.  The Handle system would then append an available numeric suffix to the chosen user handle, separated by a delimiter[.], to make it unique and store the mapping of the handle to the MSA Id in the registry.
```

Comment:
Sentences 1 per line, also "registry" does not make sense in code format. FWIW if you want code format inline, you can just use single backticks; you don't need 3 of them.
```suggestion
To achieve this, we propose the creation of a user handle registry system on Frequency chain, coupled with `MessageSourceAccount`, that allows users to choose a handle for their MSA Ids.  
The Handle system would then append an available numeric suffix to the chosen user handle, separated by a delimiter[.], to make it unique and store the mapping of the handle to the MSA Id in the registry.
```